File: sdks/csharp/src/Sudomimus.Token/WorkloadAccessTokenBody.cs
using System.Text.Json.Serialization;

namespace Sudomimus.Token;

/// <summary>Pairwise Workload actor carried by a Workload access token.</summary>
public sealed record WorkloadActor
{
    [JsonPropertyName("sub")]
    public required string Subject { get; init; }
}

/// <summary>Access-token claims for an Agent or Automation Workload.</summary>
public sealed record WorkloadAccessTokenBody
{
    [JsonPropertyName("iss")] public required string Issuer { get; init; }
    [JsonPropertyName("aud")] public required string Audience { get; init; }
    /// <summary>Owner Account's pairwise sector subject.</summary>
    [JsonPropertyName("sub")] public required string Subject { get; init; }
    [JsonPropertyName("sid")] public required string SessionId { get; init; }
    [JsonPropertyName("jti")] public required string JwtId { get; init; }
    [JsonPropertyName("iat")] public required long IssuedAt { get; init; }
    [JsonPropertyName("exp")] public required long ExpiresAt { get; init; }
    [JsonPropertyName("act")] public required WorkloadActor Actor { get; init; }
}

File: sdks/java/token/src/main/java/com/sudomimus/token/WorkloadAccessTokenBody.java
package com.sudomimus.token;

import com.fasterxml.jackson.annotation.JsonProperty;

/** Access-token payload binding an owner Account to a Workload actor. */
public final class WorkloadAccessTokenBody {
    @JsonProperty("iss") public String issuer;
    @JsonProperty("aud") public String audience;
    @JsonProperty("sub") public String subject;
    @JsonProperty("sid") public String sessionId;
    @JsonProperty("jti") public String jwtId;
    @JsonProperty("iat") public Long issuedAt;
    @JsonProperty("exp") public Long expiresAt;
    @JsonProperty("act") public WorkloadActor actor;
}
